src/screens/Server.js

```diff
@@ -69,10 +72,20 @@ class Server extends React.Component {
     }
 
     this.setState({ loading: true, errorMessage: '' });
+    const currentNetwork = hathorLib.network.getNetwork().name;
+    const currentServer = hathorLib.helpers.getServerURL();
     // Update new server in local storage
     hathorLib.wallet.changeServer(newServer);
```

Comment:
Wouldn't it be safer to change server only after the `checkApiVersion` call is resolved?

I justed tested and you are able to bypass the checkbox entirelly if you change the server when not connected to the internet.

Here are the steps to reproduce:

1. Change server to https://node1.foxtrot.testnet.hathor.network/v1a/ while disconnected from the internet
2. Do not click the "I want to connect to a full node. I understand..." checkbox
3. It will give a connection error, but will let you close the modal
4. Connect to the internet
5. You are now connected to the testnet

I don't know if a scammer would be able to convince someone to turn off the internet before changing the server, but you never know

Reply:
I agree it would be better and safer to change the server only after the `checkApiVersion`. However this would require a bigger refactor on the lib because right now all API calls are made to the server in the `localStorage`. That's why I change the server before, check the network and, in case it's a testnet, I change back to the old server.

I think our APIs could receive an optional parameter, which would be the URL to call the API but we currently don't have it.

My suggestion is to create an issue to add this optional parameter and in the meantime this problem can be handled just going back to the old server in case of promise rejection. What do you think?

Reply:
Sounds good, I agree

Reply:
Done